### PR TITLE
Fix GH#17796: Export correct system-distance

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Backport of #19512

Resolves: #17796

### CODE REVIEW

File: importexport/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -5987,9 +5987,10 @@ void ExportMusicXml::print(const Measure* const m, const int partNr, const int f
 
                         if (mpc.systemStart && !mpc.pageStart) {
                               // see System::layout2() for the factor 2 * score()->spatium()
+                              const Measure* prevSystem = mpc.prevMeasure->isMMRest() ? mpc.prevMeasure->mmRestFirst() : mpc.prevMeasure;
```
Comment (Jojo-Schmitz) [inline importexport/musicxml/exportxml.cpp:5990]: @rettinghaus: As Mu3 doesn't have `coveringMMRestOrThis()`, I'm using your earlier approach here, is that OK?
Interestingly this PR here passes the mtests, but your's doe not?!ß

Comment (rettinghaus) [reply importexport/musicxml/exportxml.cpp:5990]: Does it work? My tests didn't, that's why I changed the approach. I'd be grateful if you could help me with that test.

Comment (Jojo-Schmitz) [reply importexport/musicxml/exportxml.cpp:5990]: Well, the tests pass here

**Comment by Jojo-Schmitz** [comment]
@rettinghaus mind to check why this doesn't pass the mtest, esp. at https://github.com/Jojo-Schmitz/MuseScore/actions/runs/6372850758/job/17296046686?pr=186#step:5:1702 (the rest seems minor things to manually correct)

**Comment by rettinghaus** [comment]
That's why I asked you if this approach works for you. It _should_ work, but doesn't, that why I changed it in 4.2.0.

**Comment by Jojo-Schmitz** [comment]
Hmm, yeah, unfortunaltely Mu3 seems to be lacking the infrastructure you're using in your Mu4 PR, that `coveringMMRestOrThis()`.

**Comment by Jojo-Schmitz** [comment]
Npnsens, it doesn't, no idea why I couldn't find it initially. Thanks for getting me on track again ;-)

**Pull request merged**